config/src/main/java/org/springframework/security/config/annotation/web/builders/WebSecurity.java

```diff
@@ -395,6 +432,33 @@ public WebSecurity and() {
 			return WebSecurity.this;
 		}
 
+		/**
+		 * @param method the HttpMethod, it could be null too.
+		 * @param pathPattern the path pattern to be ignored
+		 * @since 5.5
+		 */
+		private void printWarnSecurityMessage(HttpMethod method, String pathPattern) {
```

Comment:
This warning message may need to wait as there are numerous other scenarios to cover in order for that to be complete, e.g. `regexMatchers`, `anyRequest()`, and `requestMatcher(AnyRequestMatcher.INSTANCE)`. It may be best to begin with the simplest case and do:

```java
@Override
public IgnoredRequestConfigurer anyRequest() {
	WebSecurity.this.logger.warn("Applying explicit instruction to ignore all paths");
	WebSecurity.this.logger.warn("This disables Spring Security and is not recommended");
	return super.anyRequest();
}
```

This aligns with other configured code that guards usage of `anyRequest()` but does not also attempt to parse strings to discern the intent of the application. Later on, it may be possible for the code to be more intelligent without committing to an increased public API.